crates/transaction-pool/tests/it/maintain.rs

```diff
@@ -0,0 +1,128 @@
+use reth_node_ethereum::EthereumNode;
+use reth_primitives_traits::Recovered;
+use reth_provider::CanonStateSubscriptions;
+use reth_tasks::{TaskSpawner, TokioTaskExecutor};
+use reth_transaction_pool::{
+    blobstore::InMemoryBlobStore, BlockInfo, CoinbaseTipOrdering, EthPooledTransaction,
+    PoolTransaction, TransactionOrigin, TransactionPool, TransactionPoolExt,
+};
+use std::sync::Arc;
+
+/// Helper function to create a new eth payload attributes
+pub(crate) fn eth_payload_attributes(timestamp: u64) -> EthPayloadBuilderAttributes {
+    let attributes = PayloadAttributes {
+        timestamp,
+        prev_randao: B256::ZERO,
+        suggested_fee_recipient: Address::ZERO,
+        withdrawals: Some(vec![]),
+        parent_beacon_block_root: Some(B256::ZERO),
+    };
+    EthPayloadBuilderAttributes::new(B256::ZERO, attributes)
+}
+
+/// A transaction validator that determines all transactions to be valid.
+///
+/// An actual validator impl like
+/// [TransactionValidationTaskExecutor](reth_ethereum::pool::TransactionValidationTaskExecutor)
+/// would require up to date db access.
+///
+/// CAUTION: This validator is not safe to use since it doesn't actually validate the transaction's
+/// properties such as chain id, balance, nonce, etc.
+#[derive(Debug, Default)]
+#[non_exhaustive]
+struct OkValidator;
+
+impl TransactionValidator for OkValidator {
+    type Transaction = reth_ethereum::pool::EthPooledTransaction;
+
+    async fn validate_transaction(
+        &self,
+        _origin: TransactionOrigin,
+        transaction: Self::Transaction,
+    ) -> TransactionValidationOutcome<Self::Transaction> {
+        // Always return valid
+        TransactionValidationOutcome::Valid {
+            balance: *transaction.cost(),
+            state_nonce: transaction.nonce(),
+            transaction: ValidTransaction::Valid(transaction),
+            propagate: false,
+        }
+    }
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn maintain_txpool_commit() {
+    let genesis: Genesis = serde_json::from_str(include_str!("../assets/genesis.json")).unwrap();
+    let (mut nodes, _tasks, wallet) = setup::<EthereumNode>(
+        1,
+        Arc::new(ChainSpecBuilder::mainnet().genesis(genesis).build()),
+        false,
+        eth_payload_attributes,
+    )
+    .await
+    .unwrap();
+
+    let mut node = nodes.pop().unwrap();
+
+    let executor = TokioTaskExecutor::default();
+    let txpool = reth_ethereum::pool::Pool::new(
+        OkValidator::default(),
+        CoinbaseTipOrdering::default(),
+        InMemoryBlobStore::default(),
+        Default::default(),
+    );
+
+    executor.spawn_critical(
+        "txpool maintenance task",
+        reth_transaction_pool::maintain::maintain_transaction_pool_future(
+            node.inner.provider.clone(),
+            txpool.clone(),
+            node.inner.provider.clone().canonical_state_stream(),
+            executor.clone(),
```

Comment:
hmm, for this test, we likely just need a way to mock the subscription?

sow e can ditch the entire node setup?